### PR TITLE
HARP-8521: Move text element overlay to TextElementsRenderer.

### PR DESCRIPTION
**Pull request by atomicsulfate**
Signed-off-by: Andres Mandado <andres.mandado-almajano@here.com>

Thank you for contributing to harp.gl!

Before requesting a pull request, please remember to check the following documents:
* [contribution guidelines](https://github.com/heremaps/harp.gl/blob/master/CONTRIBUTING.md)
* [coding style](https://github.com/heremaps/harp.gl/blob/master/CODINGSTYLE.md)

If you are adding new functionality we would highly appreciate if you can describe what is the capability you are adding and even better if you can add some examples. Please also remember to add tests for it.

# CI Check

Our bots will check whether your PR can be directly integrated into the mainline. We have some internal integration tests running on the background, our bots will inform you of the next steps and someone from our team will take a look and help if needed!

And please do not forget to sign-off your commit! You can read more about DCO [here](https://julien.ponge.org/blog/developer-certificate-of-origin-versus-contributor-license-agreements/). But, in short, you just need to use `git commit -s` or append `--signoff` when you are committing to the repo.

Happy contributing!

### CODE REVIEW

File: @here/harp-mapview/test/stubElevationProvider.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+export type ElevationProviderStub = sinon.SinonStubbedInstance<ElevationProvider>;
+
+/**
+ * Creates an ElevationProvider stub.
+ * @param sandbox Sinon sandbox to keep track of created stubs.
+ * @returns ElevationProvider stub.
+ */
+export function stubElevationProvider(sandbox: sinon.SinonSandbox): ElevationProviderStub {
+    const stub = sandbox.createStubInstance(ElevationProviderFake);
+    stub.getHeight.callsFake(geoPoint => {
```
Comment (nzjony) [inline @here/harp-mapview/test/stubElevationProvider.ts]: do you need this, why not provide a fake implementation in the ElevationProviderFake?

File: @here/harp-mapview/test/stubProjection.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    ProjectionType
+} from "@here/harp-geoutils";
+import * as THREE from "three";
+
+/**
+ * Creates a stub of the Mercator projection.
+ * @param sandbox Sinon sandbox to keep track of created stubs.
+ * @returns MercatorProjection stub.
+ */
+export function stubMercatorProjection(sandbox: sinon.SinonSandbox): Projection {
```
Comment (nzjony) [inline @here/harp-mapview/test/stubProjection.ts]: why do you stub the projection? Why not use mercator, or even just the identity projection?

File: @here/harp-mapview/test/stubElevationProvider.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        return geoPoint.altitude ?? 0;
+    });
+    stub.sampleHeight.callsFake(geoPoint => {
```
Comment (nzjony) [inline @here/harp-mapview/test/stubElevationProvider.ts]: as above

File: @here/harp-mapview/test/TextElementsRendererTestFixture.ts [hunk 4]
```diff
@@ -316,6 +340,24 @@ export class TestFixture {
...
     }
 
+    private setTerrainTiles(indices: number[]) {
+        expect(this.m_viewState.elevationProvider).not.undefined;
+        this.m_elevationProviderStub.getDisplacementMap.resetBehavior();
+        this.m_elevationProviderStub.getDisplacementMap.returns(undefined);
+
+        for (const index of indices) {
+            const tileKey = this.m_allTiles[index].tileKey;
+            this.m_elevationProviderStub.getDisplacementMap
```
Comment (nzjony) [inline @here/harp-mapview/test/TextElementsRendererTestFixture.ts]: do the arguments matter? Why not just .getDisplacementMap.returns(...) ?

File: @here/harp-mapview/test/stubElevationProvider.ts (hunk 1 above)
Comment (atomicsulfate) [reply @here/harp-mapview/test/stubElevationProvider.ts]: Returning a stub is more flexible, it allows users to overwrite the behaviour with whatever they want.
Writing a fake with hardcoded behaviour would only be useful for my test.
In theory I wouldn't have to write the Fake class at all, but sadly sinon doesn't let you stub an interface directly, only classes.

File: @here/harp-mapview/test/stubProjection.ts (hunk 2 above)
Comment (atomicsulfate) [reply @here/harp-mapview/test/stubProjection.ts]: good point, the identity projection is enough for what I need.

File: @here/harp-mapview/test/TextElementsRendererTestFixture.ts (hunk 4 above)
Comment (atomicsulfate) [reply @here/harp-mapview/test/TextElementsRendererTestFixture.ts]: the test is simulating how the terrain tiles become available in the different frames.
At the beginning of a frame, that function "terrainTiles()" is called with the indices of the terrain tiles that should be available at that moment. For those available, getDisplacementMap must return the displacement map. For any other tiles that function returns undefined, as it's defined just above that loop.

File: @here/harp-mapview/test/stubElevationProvider.ts (hunk 1 above)
Comment (atomicsulfate) [reply @here/harp-mapview/test/stubElevationProvider.ts]: besides, getDisplacementMap stub is configured in the test to return different results depending on the passed tile key (see my answer below).

File: @here/harp-mapview/test/stubElevationProvider.ts (hunk 3 above)
Comment (atomicsulfate) [reply @here/harp-mapview/test/stubElevationProvider.ts]: see my answer above.

File: @here/harp-mapview/test/TextElementsRendererTestFixture.ts (hunk 4 above)
Comment (nzjony) [reply @here/harp-mapview/test/TextElementsRendererTestFixture.ts]: ok, makes sense.

File: @here/harp-mapview/test/stubElevationProvider.ts (hunk 1 above)
Comment (nzjony) [reply @here/harp-mapview/test/stubElevationProvider.ts]: getDisplacementMap makes sense, and it makes sense to return a stub, I'm more interested to know why you need to have callsFake for getHeight and sampleHeight, couldn't they just be in the fake class?

Comment (atomicsulfate) [reply @here/harp-mapview/test/stubElevationProvider.ts]: createStubInstance() creates a fully stubbed instance of the ElevationProviderFake, so none of the method implementations are preserved. That's why I set the behaviour later with callsFake().

**Pull request merged**